kava/client.go

```diff
@@ -176,3 +164,56 @@ func (c *Client) Balance(
 		Balances: balances,
 	}, nil
 }
+
+// Block returns rosetta block for an index or hash
+func (c *Client) Block(
+	ctx context.Context,
+	blockIdentifier *types.PartialBlockIdentifier,
+) (*types.BlockResponse, error) {
+	block, err := c.getBlockResult(blockIdentifier)
+	if err != nil {
+		return nil, err
+	}
+
+	height := block.Block.Header.Height
+	identifier := &types.BlockIdentifier{
+		Index: height,
+		Hash:  block.BlockID.Hash.String(),
+	}
+
+	var parentIdentifier *types.BlockIdentifier
+	if height == 1 {
+		parentIdentifier = identifier
+	} else {
+		parentIdentifier = &types.BlockIdentifier{
+			Index: height - 1,
+			Hash:  block.Block.Header.LastBlockID.Hash.String(),
+		}
+	}
+
+	return &types.BlockResponse{
+		Block: &types.Block{
+			BlockIdentifier:       identifier,
+			ParentBlockIdentifier: parentIdentifier,
+			Timestamp:             block.Block.Header.Time.UnixNano() / int64(1e6),
+			Transactions:          []*types.Transaction{},
+		},
+	}, nil
+}
+
+func (c *Client) getBlockResult(blockIdentifier *types.PartialBlockIdentifier) (block *ctypes.ResultBlock, err error) {
+	switch {
+	case blockIdentifier == nil:
+		block, err = c.rpc.Block(nil)
```

Comment:
Just curiousity, what does passing nil to `rpc.Block` result in?

Reply:
It'll return the latest height. It's a requirement for the /account/balance endpoint, not for the /block endpoint.

Rpc reference is here https://github.com/tendermint/tendermint/blob/master/rpc/core/blocks.go#L92

Reply:
Added some comments to make that clear in the code